File: C++ solutions/646. Maximum Length of Pair Chain.cpp
class Solution {
public:
    int findLongestChain(vector<vector<int>>& pairs) {
        if(pairs.size() ==0 ) return 0;
        sort(pairs.begin(), pairs.end());

        int curr=INT_MAX,res=0;
    
        for(int i=pairs.size()-1;i>=0;i--)
        {
            if(pairs[i][1]<curr)
            {
                res++;
                curr=pairs[i][0];
            }
        }
        return res;
    }
};

File: JavaScript solutions/646. Maximum Length of Pair Chain.js
/**
 * @param {number[][]} pairs
 * @return {number}
 */
var findLongestChain = function (pairs) {
  if (pairs.length === 0) return 0;
  pairs.sort((a, b) => a[0] - b[0]);

  let curr = Number.MAX_SAFE_INTEGER;
  let res = 0;

  for (let i = pairs.length - 1; i >= 0; i--) {
    if (pairs[i][1] < curr) {
      res++;
      curr = pairs[i][0];
    }
  }
  return res;
};
